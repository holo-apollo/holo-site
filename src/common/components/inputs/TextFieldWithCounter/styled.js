// @flow
import styled, { css } from 'styled-components';

import palette from 'common/palette';

const getHelperStyles = ({ hasHelperText }: { hasHelperText: boolean }) => css`
  justify-content: ${hasHelperText ? 'space-between' : 'flex-end'};
`;

export const HelperCont = styled.span`
  width: 100%;
  display: flex;
  align-items: center;

  ${getHelperStyles};
`;

export const CounterCont = styled.span`
  color: ${palette.grey};
`;
