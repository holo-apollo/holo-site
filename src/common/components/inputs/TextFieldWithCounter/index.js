// @flow
import * as React from 'react';

import TextField from 'common/components/inputs/TextField';
import useInput from 'common/hooks/useInput';
import { HelperCont, CounterCont } from './styled';

type Props = {
  onChange?: (SyntheticInputEvent<HTMLInputElement>) => void,
  maxLength: number,
  helperText?: React.Node,
};

const TextFieldWithCounter = (props: Props) => {
  const { value, onChange } = useInput('');

  function handleInputChange(event: SyntheticInputEvent<HTMLInputElement>) {
    onChange(event);
    props.onChange && props.onChange(event);
  }

  const helperText = (
    <HelperCont hasHelperText={Boolean(props.helperText)}>
      {props.helperText}
      <CounterCont>
        {value.length}/{props.maxLength}
      </CounterCont>
    </HelperCont>
  );

  return (
    <TextField
      {...props}
      value={value}
      onChange={handleInputChange}
      helperText={helperText}
    />
  );
};

export default TextFieldWithCounter;
