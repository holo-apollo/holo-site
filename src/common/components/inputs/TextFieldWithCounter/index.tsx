import React, { ChangeEvent } from 'react';

import TextField from 'common/components/inputs/TextField';
import useInput from 'common/hooks/useInput';
import { HelperCont, CounterCont } from './styled';

type Props = React.ComponentProps<typeof TextField>;

const TextFieldWithCounter = (props: Props) => {
  const { value, onChange } = useInput('');

  function handleInputChange(event: ChangeEvent<HTMLInputElement>) {
    onChange(event);
    props.onChange && props.onChange(event);
  }

  const helperText = (
    <HelperCont hasHelperText={Boolean(props.helperText)}>
      {props.helperText}
      <CounterCont>
        {value.length}/{props.maxLength}
      </CounterCont>
    </HelperCont>
  );

  return (
    <TextField
      {...props}
      value={value}
      onChange={handleInputChange}
      helperText={helperText}
    />
  );
};

export default TextFieldWithCounter;
