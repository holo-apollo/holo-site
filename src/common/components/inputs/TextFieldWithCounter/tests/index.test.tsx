import React from 'react';
import renderer from 'react-test-renderer';

import TextFieldWithCounter from '../index';

describe('TextFieldWithCounter', function() {
  it('should match snapshot', function() {
    const tree = renderer
      .create(
        <TextFieldWithCounter name="name" label="Label" maxLength={100} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
