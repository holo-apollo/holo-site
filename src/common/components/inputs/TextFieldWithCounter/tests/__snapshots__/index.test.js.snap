// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldWithCounter should match snapshot 1`] = `
<div
  className="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
  maxLength={100}
>
  <label
    className="MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
    data-shrink={false}
    style={
      Object {
        "fontSize": "12px",
      }
    }
  >
    Label
  </label>
  <div
    className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-fullWidth-45 MuiInput-fullWidth-30 MuiInputBase-formControl-37 MuiInput-formControl-24"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input-46 MuiInput-input-31"
      disabled={false}
      maxLength={100}
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
  </div>
  <p
    className="MuiFormHelperText-root-53"
    style={
      Object {
        "fontSize": "10px",
      }
    }
  >
    <span
      className="styled__HelperCont-pe4d84-0 bjpMQO"
    >
      <span
        className="styled__CounterCont-pe4d84-1 cmBwEd"
      >
        0
        /
        100
      </span>
    </span>
  </p>
</div>
`;
